event_routing_backends/backends/events_router.py

```diff
@@ -173,6 +186,24 @@ def send(self, event):
                         host['host_configurations'],
                     )
 
+    def queue_event(self, event):
+        """
+        Queue the event to be sent to configured routers.
+
+        """
+        redis = get_redis_connection()
+
+        event["timestamp"] = event["timestamp"].isoformat()
+        redis.lpush(self.queue_name, json.dumps(event))
+        queue_size = redis.llen(self.queue_name)
+
+        logger.info(f'Event {event["name"]} has been queued for batching.')
+        if queue_size < settings.EVENT_ROUTING_BACKEND_BATCH_SIZE:
```

Comment:
I think that since the consumer is a single process you can declare a global in this file to manage `LAST_SUCCESSFUL_SEND_TIME` and use it to determine when to send even if the list isn't big enough yet. It still won't trigger unless an event arrives, but is probably better than nothing or adding a bunch of complicated threading. If someone has that low of traffic they can always set the batch size to 1 or turn this off.